instrumentation-genai/opentelemetry-instrumentation-vertexai/src/opentelemetry/instrumentation/vertexai/utils.py

```diff
@@ -137,6 +150,21 @@ def get_genai_request_attributes(
     return attributes
 
 
+def get_genai_response_attributes(
+    response: prediction_service.GenerateContentResponse
+    | prediction_service_v1beta1.GenerateContentResponse,
+) -> dict[str, AttributeValue]:
+    finish_reasons: list[str] = [
+        candidate.finish_reason.name for candidate in response.candidates
```

Comment:
Curious, do we know the values of the what the `FinishReason` [enum](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/GenerateContentResponse#FinishReason) is?

Reply:
Maybe we should call `lower()` on the reasons similar to how we do it in `map_finish_reasons()`?

Reply:
> Curious, do we know the values of the what the `FinishReason` [enum](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/GenerateContentResponse#FinishReason) is?

I'm not sure what you mean, the doc you linked has the possible values



> Maybe we should call `lower()` on the reasons similar to how we do it in `map_finish_reasons()`?

I'm actually not confident we should be doing any re-mapping. I only made it lower in map_finish_reasons to match the OTel semconv enum. Want to discuss more before updating

Reply:
Updated to use `_map_finish_reason()` as well